### DISCUSSION TITLE
Call custom pmd apex rules from SonarLint VS Code plugin

### DISCUSSION THREAD

**Discussion by balbhadra9** [pmd/pmd#3020] [Q&A] [answered]
Hi,
We have written a few PMD apex custom rules. At present we run pmd batch file from the command line to generate the violation report file.
Most of the developers at our company use Sonarlint as VSCode extension. Is there a way to configure Sonarlint so that it calls pmd batch file automatically and violations get reported in the VS code UI?

thanks,

**Comment by adangel** [comment, marked as answer]
I would recommend to use https://github.com/ChuckJonas/vscode-apex-pmd - that's the vscode Apex PMD plugin. It seems to support custom rules as well.
